ostd/src/arch/x86/mm/mod.rs

```diff
@@ -302,3 +330,20 @@ impl fmt::Debug for PageTableEntry {
             .finish()
     }
 }
+
+/// Programs the PAT MSR so that write-combining mappings use the correct memory type.
+pub(crate) fn configure_pat() {
+    let mut programmed_pat = 0u64;
+    for (idx, policy) in IA32_PAT_MAPPINGS.iter().copied().enumerate() {
+        programmed_pat |= (policy.to_ia32_pat_memory_type_entry() as u64) << (idx * 8);
+    }
+
+    // SAFETY: Writing `IA32_PAT` merely programs the page attribute table MSR.
+    // This code runs on the bootstrap processor before other CPUs depend on
```

Comment:
This is not true. The code will also run in APs.
---

To make things simpler, I think most operations in `enable_cpu_features` is safe just because it builds up our global invariant of the CPU state. We'd better figure out a better way to write safety comments.